src/states/adult/mod.rs

```diff
@@ -235,20 +235,18 @@ impl Adult {
     fn dispatch_routing_message(
         &mut self,
         msg: SignedRoutingMessage,
-        _outbox: &mut dyn EventBox,
-    ) -> Result<(), RoutingError> {
-        self.add_message_to_backlog(msg);
-        Ok(())
-    }
-
-    // Backlog the message to be processed once we are established.
-    fn add_message_to_backlog(&mut self, msg: SignedRoutingMessage) {
-        trace!(
-            "{} Not elder yet. Delaying message handling: {:?}",
-            self,
-            msg
-        );
-        self.routing_msg_backlog.push(msg)
+    ) -> Result<Transition, RoutingError> {
+        match &msg.routing_message().content {
+            MessageContent::GenesisUpdate(info) => {
+                self.check_signed_message_trust(&msg)?;
+                self.handle_genesis_update(info.clone())
+            }
+            _ => {
+                self.routing_msg_backlog.push(msg.clone());
+                self.send_signed_message_to_elders(msg)?;
```

Comment:
That does not seem right, dispatch_routing_message did not do the send_signed_message in either Elder or Adult. I don't think it should now.

Reply:
Why is that a problem?

Reply:
Ok, I see. It's a bit inconsistent. I'll change it.

Reply:
Level of abstraction, the level above does the resending, dispatch is to handle message we need to address. It makes is clearer in the calling function that resending is happening as needed. (Here we split it in 2, part of resending is done in calling function.

This will be even worth with work I'm looking at to avoid reserialization, it would force us to pass to the dispatch function what is needed for the resend.